group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -247,31 +324,78 @@ GroupMetadataManager build() {
      */
     private MetadataImage metadataImage;
 
+    /**
+     * An empty result returned to the state machine. This means that
+     * there are no records to append to the log.
+     *
+     * Package private for testing.
+     */
+    static final CoordinatorResult<Void, Record> EMPTY_RESULT =
+        new CoordinatorResult<>(Collections.emptyList(), CompletableFuture.completedFuture(null));
```

Comment:
nit: I was wondering whether it would make sense to move this to `CoordinatorResult`. We could have a static public constant called `EMPTY`. Then, we could use `CoordinatorResult.EMPTY` in the code. This is a pattern that we already use in a few other places. What do you think?

Reply:
that forces the CoordinatorResult class to become non-generic which i don't think we want.

Reply:
ah.. did not think about that.